.github/workflows/gradle.yml

```diff
@@ -0,0 +1,36 @@
+name: Gradle CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: gradle
+    
+    - name: Grant execute permission for gradlew
```

Comment:
Let's use https://github.com/gradle/actions/tree/main/setup-gradle instead because it sets up caching etc